### DISCUSSION TITLE
Report | Week 17

### DISCUSSION THREAD

**Discussion by tech-n1c** [SE-TINF22B6/time2shine#91] [Show and tell]
Hey everyone,

this week we did some refactoring. We used the Google Lighthouse tool, which was already presented in class. It showed us, that we have a lot of overhead in our frontend, which is caused by redundant TypeScript files that we used for the Page Layouts. Since we use the same layout for all pages, it is completely unnecessary to use the same layout file for each Sub-Page. 

This way, we could reduce the data transferred to a user when opening our web page.

---
![refactor](https://github.com/SE-TINF22B6/time2shine/assets/59262249/9bc3ae2a-9bed-4b51-ac97-35b9decf42cb)

**Comment by nuo-li** [comment]
Well done. Any other clean code principles your team has practiced?

**Comment by mirak55** [comment]
Nice work on your task about clean code, you also provided good documentation, on how you implemented it in your own project. Keep up the good work, and looking forward, to see some new progress on your website itself.

**Comment by BenjaminB109** [comment]
Dear time2shine Team,
good work eliminating the redundancy and thanks for mentioning the Google Lighthouse tool if anyone wants to use it. Keep up the good work.

**Comment by maxschwinghammer** [comment]
Dear time2shine team,
using Google Lighthouse to cut redundant TypeScript files is a smart optimization. Nice job on reducing data transfer and improving performance!
Your team DHBWhub

**Comment by fehlend0** [comment]
Dear time2shine team,

it was a great idea to use Google Lighthouse to identify areas for improvement. Reducing redundant TypeScript files by streamlining your Page Layouts is a smart decision - this not only optimizes your code, but also enhances the user experience by minimizing data transfer.  

Good job!

Your Statify Team
